src/userobjects/OpenMCCellTransform.C

```diff
@@ -0,0 +1,132 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCCellTransform.h"
+#include "CardinalAppTypes.h"
+#include <string>
+
+registerMooseObject("CardinalApp", OpenMCCellTransform);
+
+InputParameters
+OpenMCCellTransform::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params += OpenMCBase::validParams();
+
+  params.addRequiredParam<std::vector<int32_t>>(
+      "cell_ids", "List of OpenMC cell IDs whose filled universes will be transformed.");
+
+  MooseEnum transform_type("translation rotation", "translation");
+  params.addParam<MooseEnum>("transform_type",
+                             transform_type,
+                             "Type of transform to apply: 'translation' (dx,dy,dz) or 'rotation'"
+                             "(φ, θ, ψ) in degrees.");
+
+  params.addRequiredParam<std::vector<PostprocessorName>>(
+      "transform_array",
+      "An array of three postprocessors. For translation this array expects (dx, dy, dz) in mesh"
```

Comment:
Do you want to clarify that the values don't HAVE to be postprocessors? For constants, this could be a lot more convenient than having people create a bunch of `ConstantPostprocessors`